pkg/trace/api/debugger.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/fargate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// logsIntakeURLTemplate specifies the template for obtaining the intake URL along with the site.
+	logsIntakeURLTemplate = "https://http-intake.logs.%s/v1/input"
+)
+
+// debuggerProxyHandler returns an http.Handler proxying requests to the logs intake. If the logs intake url cannot be
+// parsed, the returned handler will always return http.StatusInternalServerError with a clarifying message.
+func (r *HTTPReceiver) debuggerProxyHandler() http.Handler {
+	tags := fmt.Sprintf("host:%s,default_env:%s,agent_version:%s", r.conf.Hostname, r.conf.DefaultEnv, info.Version)
+	if orch := r.conf.FargateOrchestrator; orch != fargate.Unknown {
+		tag := fmt.Sprintf("orchestrator:fargate_%s", strings.ToLower(string(orch)))
+		tags = tags + "," + tag
+	}
+	intake := fmt.Sprintf(logsIntakeURLTemplate, config.DefaultSite)
+	if v := config.Datadog.GetString("apm_config.debugger_dd_url"); v != "" {
+		intake = v
+	} else if site := config.Datadog.GetString("site"); site != "" {
+		intake = fmt.Sprintf(logsIntakeURLTemplate, site)
+	}
+	target, err := url.Parse(intake)
+	if err != nil {
+		msg := fmt.Errorf("error parsing snapshot intake URL %s: %v", intake, err)
+		log.Critical(msg)
+		return debuggerErrorHandler(msg)
+	}
+	return newDebuggerProxy(r.conf.NewHTTPTransport(), target, r.conf.APIKey(), tags)
+}
+
+// debuggerErrorHandler always returns http.StatusInternalServerError with a clarifying message.
+func debuggerErrorHandler(err error) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		msg := fmt.Sprintf("Debugger Proxy is OFF: %v", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+	})
+}
+
+// newDebuggerProxy returns a new httputil.ReverseProxy proxying and augmenting requests with headers containing the tags.
+func newDebuggerProxy(rt http.RoundTripper, target *url.URL, key string, tags string) *httputil.ReverseProxy {
+	logger := logutil.NewThrottled(5, 10*time.Second) // limit to 5 messages every 10 seconds
+	director := func(req *http.Request) {
+		newTags := tags
```

Comment:
I think we don't need a new variable here, and can just continue using `tags` (it's already a copy of the original string if that was your concern).

Reply:
But it will live longer than each invocation of the director function 😅

Reply:
Ah yes, you're right. My bad. Carry on. Maybe we can call it `ddtags`? It will match the query string. I'm not a very big fan of camelCaseVariables that are short lived. I notice in Go they are usually very short if they only last a few lines. Would you mind?

Reply:
Not at all :+1: